Ellipsis

schew opened the pull request:
Also migrated indices template to src
Added styling for when search is shortened horizontally
- top filter (orders/datapoints) is now seen as an ellipsis when condensed 
- word wrap appears
- minwidth set for dropdown
- changed icon position in order search results when shortened to sit on the top line

you probably won't be able to see some of the real changes here since a lot of the problems cropped up with the f4 app overriding the styles here. so link with react-factoryfour
@allytart this branch would probably be a good place to adjust any styling on the search level - i can do it too if you'd like. let me know if you have problems linking

allytart commented:
@schew Added some styling edits!

nktasingh commented on src/components/results/Datapoint.jsx:
```diff
@@ -34,7 +34,7 @@ const Datapoint = ({ element, dataProps }) => {
 			onKeyPress={onDatapointClick}
 		>
-			<Icon color="accent3" className={css(style.icon)} icon="folder" />
+			<Icon color="gray3" className={css(style.icon)} icon="linearScale" />
```
Should the datapoint icon color be the same as the one used for the order icon? Or are we looking to associate this gray with datapoints... In that case should the "Datapoints" header be the same gray? Just curious!

allytart replied:
The order/job icons have colors simply because the colors correspond with whether it's a job/order (in this case, when we do have jobs the header would be "Orders/Jobs"), so the datapoint icon as the neutral gray makes most sense to me. That's how I went about it, but let me know what you think! The headers should all the same accent blue for consistency

schew replied:
i'm cool with it being a different color. should we do a brighter color since there aren't a lot of colors?

nktasingh replied:
Ah I get it! So future icons (say, a customer icon) would also be gray. Makes sense

nktasingh commented on src/components/styles/Search.js:
```diff
@@ -18,25 +19,33 @@ const styles = StyleSheet.create({
 		display: 'inline',
 		backgroundColor: '#EBF0FF',
-		fontSize: fontSize.subheading,
+		fontSize: '12px',
```
The filter text and icon don't seem to be vertically aligned?

nktasingh commented on src/components/styles/Filters.js:
```diff
@@ -1,23 +1,24 @@
-import { colors, fontSize, descendant } from '@factoryfour/react-style';
+import { colors, descendant } from '@factoryfour/react-style';
```
When I click on the filter dropdown, the check icons are hard to read? It seems like some onHover styling makes them completely white, but even when not hovering, the light gray is a little difficult to see

allytart replied:
Noticed the hover white color as well, but wasn't sure if this was intentional! @schew 

Good point on the light gray though, we can bump it up to gray3

schew replied:
there's a hover filter that's placed on the dropdown that sets the brightness to 110%, but it's getting inherited by the children and lightening them. i went ahead and removed the brightness filter and converted it to the accent1 color with 10% brightness baked into the hex code.

allytart commented:
Noticing that the vertical alignment of the navbar content is slightly off :(

![image](https://user-images.githubusercontent.com/14914915/36443923-20cf6bf2-1648-11e8-9ff1-49cfe21f30a1.png)

schew commented:
- Fixed the margins for the search bar in the navbar content. The breadcrumb can be fixed when the react-search version is bumped in react-factoryfour
- Changed the icon color in the filter option
- Fixed the hover behavior over the filter option

allytart commented:
@nktasingh @schew What are your thoughts on keeping the dropdown open after a filter is selected, and close only when the user clicks out of it? It may make it easier to select multiple filters that way

schew commented:
Good idea!

nktasingh commented:
@schew Weird, I'm not seeing the changes above! We can talk about it in person :)

nktasingh commented:
@allytart I was also wondering, should there be some sort of onHover styling when clicking to remove "x Orders" in the search bar?

allytart commented:
@schew Regarding the hover behavior nikita was talking about above ^, currently the label sits inside of the search bar and the search bar padding extends beyond the label, so it would look strange with any hover background change. If we could swap the edge padding (with the rounded borders) into the label block, we can add a background hover color transition to the label. 

![image](https://user-images.githubusercontent.com/14914915/36488843-79d325e0-16f2-11e8-95f8-1627e449c4de.png)

allytart commented:
@schew Some minor styling! :) For the inline filter tag can we have the background white and the text and icon color colors.alert, and on hover just the inverted colors? 

I saw that when hovering over the filter buttons, the blue is different from the default accent1. Realized that hover hue-rotate filter (which looks odd for buttons aside from the task "nodes") is in the button component. For a future pr, is it possible to move that out so it's just styling for the task node buttons? Or maybe have it as a prop?

schew commented:
Ready for another look! i dare you to break it.

allytart commented:
@schew Everything looks and works great!!! I added some minor things yesterday :) One last thing to think about before merge--the height of the search/breadcrumb bar seems to take up quite a bit of the page. @nktasingh @schew thoughts on shrinking it just a bit?
